datafusion/expr/src/udwf.rs

```diff
@@ -281,6 +285,10 @@ pub trait WindowUDFImpl: Debug + Send + Sync {
 
     /// What [`DataType`] will be returned by this function, given the types of
     /// the arguments
+    #[deprecated(
+        since = "41.0.0",
+        note = "Use `field` instead to define the final result of evaluating this user-defined window function."
+    )]
```

Comment:
Maybe remove this right now instead of marking it as deprecated?

I checked and the deprecated warning does not show for trait implementations. It is only shown as a warning during cargo build and an error in clippy when the trait method is called.